src/JSHandle.ts

```diff
@@ -442,10 +455,18 @@ export class ElementHandle extends JSHandle {
    * @return {!Promise<!Array<!ElementHandle>>}
    */
   async $$(selector: string): Promise<ElementHandle[]> {
-    const arrayHandle = await this.evaluateHandle(
-        (element, selector) => element.querySelectorAll(selector),
-        selector
-    );
+    let handler = (element: Element, selector: string): ElementHandle[] => element.querySelectorAll(selector) as unknown as ElementHandle[];
+    if (selector.indexOf(SEARCH_SEPARATOR) !== -1) {
```

Comment:
Can we extract this common `indexOf` + `split` logic?

Reply:
Done.

Reply:
Instead of just `_getQueryFunctionNameAndSelector`, could we extract the entire block? It's shared between `$$` and `$$eval` below.

```js
let handler = (element: Element, selector: string): JSHandle => Array.from(element.querySelectorAll(selector)) as unknown as JSHandle;
        (element, selector) => Array.from(element.querySelectorAll(selector)),	    if (selector.indexOf(SEARCH_SEPARATOR) !== -1) {
        selector	      const {name, updatedSelector} = this._getQueryFunctionNameAndSelector(selector);
    );	      const queryFunction = customQueryHandlers().get(name);
      if (!queryFunction)
        throw new Error(`Query set to use "${name}", but no query function of that name was found`);

      handler = queryFunction as typeof handler;
      selector = updatedSelector;
    }
```

Could be something like:

```js
const {handler, selector} = getHandlerAndSelector();
```

(with possibly better names)

WDYT?